overlord/managers_test.go

```diff
@@ -653,6 +654,54 @@ apps:
 	c.Assert(s.automaticSnapshots, DeepEquals, []automaticSnapshotCall{{"foo", map[string]interface{}{"key": "value"}, nil}})
 }
 
+func (s *mgrsSuite) TestHappyRemoveWithQuotas(c *C) {
+	oldEstimateSnapshotSize := snapstate.EstimateSnapshotSize
+	snapstate.EstimateSnapshotSize = func(st *state.State, instanceName string, users []string) (uint64, error) {
+		return 0, nil
+	}
+	defer func() {
+		snapstate.EstimateSnapshotSize = oldEstimateSnapshotSize
+	}()
```

Comment:
You can just pass `&snapstate.RemoveFlags{Purge: true}` to Remove, avoiding the need for this mock (automatic snapshots are tested elsewhere in more targeted tests).

Reply:
thanks for the tip!